Processing vector geometry improvements

ghtmtt opened the pull request:
fix #2563 fix #2591 and other small changes

some pictures added, actually many pictures.. I think it is way better to have clear images and less description (a good balance of both of them). Users can get the idea faster IMHO

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -518,6 +519,77 @@ Output
+  value depending on the feature attribute).
+
+``Wedge width (in degrees)`` [number]
+  Width (in degrees) of the buffer. It can be a unique value (same value for all
```
`Angle (in degrees) of the wedge shaped buffer`

ghtmtt replied:
?

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -2228,6 +2300,73 @@ Outputs
+Output
+......
+
+``Buffered layer`` [vector: polygon]
```
~~layer~~

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -2228,6 +2300,73 @@ Outputs
+
+Tapered buffers |32|
+--------------------
+Creates tapered buffers along line geometries, using a specified start and end
```
Given that it creates one buffer per feature, better use `tapered buffer` instead.

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -2228,6 +2300,73 @@ Outputs
+  Input line vector layer.
+
+``Start width`` [number]
+  Starting width of the tapered buffer. It can be a unique value (same value for
```
`Represents the radius of the buffer applied at the start point (or first vertex?) of the line feature.` or something like that

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -2228,6 +2300,73 @@ Outputs
+  on the feature attribute).
+
+``End width`` [number]
+  Ending width of the tapered buffer. It can be a unique value (same value for
```
As above?

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -518,6 +519,77 @@ Output
+``Azimuth (degrees from North)`` [number]
+  Angle (in degrees) for the middle of the wedge to point. It can be a unique value
+  (same value for all the features) or it can be taken from features data (different
+  value depending on the feature attribute).
```
note: if the DD issue gets agreement, it should be applied to other places so for readability, I ignore the remark in the other reviews. and maybe some changes may be deferred until that agreement...

ghtmtt replied:
Ok I made a first layout example with `[number or data defined icon_here]`. Looks nice ;)

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -2228,6 +2300,73 @@ Outputs
+
+See also
+........
+:ref:`qgisbufferbym`, :ref:`qgisbuffer`
```
I'd also add the wedge one

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -2317,6 +2456,40 @@ Outputs
+Output
+......
+
+``Buffered layer`` [vector: polygon]
```
still no layer

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -2317,6 +2456,40 @@ Outputs
+.. figure:: img/variable_buffer_m.png
+   :align: center
+
+   Tapered buffer example
```
tapered?

DelazJ commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -2228,6 +2300,73 @@ Outputs
+......
+
+``Buffered layer`` [vector: polygon]
+  Variable buffer polygon layer
```
could be nice to have the idea of tapered here instead.

ghtmtt commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -518,6 +519,77 @@ Output
+
+Create wedge buffers |32|
+-------------------------
+Creates wedge shaped buffers from input points.
```
same for me. I'll leave this small description as it is (for now) and maybe enhance it in a future

ghtmtt commented on source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst:
```diff
@@ -518,6 +519,69 @@ Output
+``Input layer`` [vector: point]
+  Input point vector layer
+
+``Azimuth (degrees from North)`` [number or data-defined |dataDefined|]
```
mmm I think that the discussion here could become really long :) 

for the moment I removed the `data-defined` *string* and left only the icon.

> whether the option is data-defined or not, the value returned should still be a number to be accepted so [number] is always the right data type.

agreed

> I think that adding the icon should not anyway exempt us from specifying for each concerned parameter description that it's a unique or data-defined value (made a wording in an earlier comment in this PR)

here I'm -0.5 :) just kidding.. but the problem to specify that this parameter can also accept data-defined values is that we can have a lot of repetition in the algorithm description (like in the first commit of this PR). While having just the icon could be too implicit and uses cannot be aware of it.

I don't know how we can solve this...

ghtmtt commented:
@DelazJ thanks for all the comments. I removed some pictures fro the wedge buffer and replaced by only  2: a general picture (with some weird shape) and another one explaining the azimuth and width parameters

ghtmtt commented:
@DelazJ Nope **you** where right, see screenshot;) of course I have a lot of layers in the project (point, line, polygon...)

if you give me the OK I will resolve the conflicts and merge this PR. Thanks!

![create wedge buffers_106](https://user-images.githubusercontent.com/2884884/42459265-a873bd04-839b-11e8-8ea0-6abfe339319e.png)

DelazJ commented:
Sorry but **YOU WERE RIGHT**. The screenshot you are showing has nothing to do with the 
384f5f9  commit. That commit concerns "Swap X,Y coordinates" alg  that takes all types of geometry and not only point while you are showing wedge buffer screenshot.

ghtmtt commented:
ahhh.. sorry.. wrong algorithm.. So I was right :tada: ;) description changed.. I'll fix the conflicts and merge the PR. Thanks again @DelazJ
